File: python/Medium/SingleNumber.py
__author__ = 'Lei Chen'
'''
Given an array of integers, every element appears twice except for one. Find that single one.

Note:
Your algorithm should have a linear runtime complexity. Could you implement it without using extra memory?
'''

class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ret = 0
        for n in nums:
            ret = ret ^ n
        return ret

File: java/SingleNumber.java
public class SingleNumber {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		int[] input = new int[] { 5, 1, 3, 4, 4, 1, 5 };
		System.out.println(singleNumber(input));
	}

	public static int singleNumber(int[] A) {
		int ret = 0;
		for (int i = 0; i < A.length; i++) {
			ret = ret ^ A[i];
		}
		return ret;
	}

}
